src/CoreMedia/CoreMedia.cs

```diff
@@ -87,22 +90,24 @@ public struct CMTimeRange {
 		[iOS (9,0)]
 		[Mac (10,11)]
 #endif
-		public static NSString TimeMappingTargetKey { get; private set; }
+		public static NSString? TimeMappingTargetKey { get; private set; }
 
 		static CMTimeRange () {
 			var lib = Libraries.CoreMedia.Handle;
 			var retZero = Dlfcn.dlsym (lib, "kCMTimeRangeZero");
-			Zero = (CMTimeRange)Marshal.PtrToStructure (retZero, typeof(CMTimeRange));
+			Zero = (CMTimeRange)Marshal.PtrToStructure (retZero, typeof(CMTimeRange))!;
 
 			var retInvalid = Dlfcn.dlsym (lib, "kCMTimeRangeInvalid");
 #if !XAMCORE_3_0
-			Invalid = (CMTimeRange)Marshal.PtrToStructure (retInvalid, typeof(CMTimeRange));
+#pragma warning disable CS0618 // Type or member is obsolete
+			Invalid = (CMTimeRange)Marshal.PtrToStructure (retInvalid, typeof(CMTimeRange))!;
+#pragma warning restore CS0618 // Type or member is obsolete
 #endif
-			InvalidRange = (CMTimeRange)Marshal.PtrToStructure (retInvalid, typeof(CMTimeRange));
+			InvalidRange = (CMTimeRange)Marshal.PtrToStructure (retInvalid, typeof(CMTimeRange))!;
 
 			var retMappingInvalid = Dlfcn.dlsym (lib, "kCMTimeMappingInvalid");
-			if (retMappingInvalid  != IntPtr.Zero)
-				InvalidMapping = (CMTimeRange)Marshal.PtrToStructure (retMappingInvalid, typeof(CMTimeRange));
+			if (retMappingInvalid != IntPtr.Zero)
+				InvalidMapping = (CMTimeRange)Marshal.PtrToStructure (retMappingInvalid, typeof(CMTimeRange))!;
```

Comment:
I was also not sure if these castings should be `!`. I tried making these CMTimeRange nullable, but I remember Rolf said not to make structs nullable

Reply:
For both of your questions, I would use the ! since we might be changing behaviour in both cases.